application/src/main/java/org/thingsboard/server/service/telemetry/InternalTelemetryService.java

```diff
@@ -23,14 +23,16 @@
 import org.thingsboard.rule.engine.api.TimeseriesSaveRequest;
 import org.thingsboard.server.common.data.kv.TimeseriesSaveResult;
 
+import java.util.List;
+
 /**
  * Created by ashvayka on 27.03.18.
  */
 public interface InternalTelemetryService extends RuleEngineTelemetryService {
 
     ListenableFuture<TimeseriesSaveResult> saveTimeseriesInternal(TimeseriesSaveRequest request);
 
-    void saveAttributesInternal(AttributesSaveRequest request);
+    ListenableFuture<List<Long>> saveAttributesInternal(AttributesSaveRequest request);
```

Comment:
Let's wrap it into AttributesSaveResult, similar to TimeseriesSaveResult

Reply:
This is a nice improvement, added it to this PR